core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java

```diff
@@ -127,6 +131,7 @@ public TaskInfoFetcher(
         this.spanBuilderFactory = requireNonNull(spanBuilderFactory, "spanBuilderFactory is null");
         this.stats = requireNonNull(stats, "stats is null");
         this.estimatedMemory = requireNonNull(estimatedMemory, "estimatedMemory is null");
+        this.retryPolicy = retryPolicy;
```

Comment:
`requireNonNull(retryPolicy, "retryPolicy is null")`

Reply:
Done 304b3311b8